extensions/brand-fetch/src/index.tsx

```diff
@@ -0,0 +1,102 @@
+import { Form, ActionPanel, Action, showToast, getPreferenceValues, Toast, Grid, Clipboard, Icon } from "@raycast/api";
+import { useEffect, useState } from "react";
+import { retrieveBrand } from "./source";
+import { RetrieveBrandResponse } from "./source.types";
+import { homedir } from "os";
+import axios from "axios";
+import { writeFile } from "fs/promises";
+import { join } from "path";
+import { randomUUID } from "crypto";
+
+const DOWNLOADS_DIR = `${homedir()}/Downloads`;
+
+type Values = {
+  textfield: string;
+};
+
+interface Preferences {
+  bfApiKey: string;
+}
+
+export default function Command() {
+  const { bfApiKey } = getPreferenceValues<Preferences>();
+  const [error, setError] = useState<string | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [resp, setResp] = useState<RetrieveBrandResponse | undefined>();
+
+  useEffect(() => {
+    if (error) {
+      showToast({
+        style: Toast.Style.Failure,
+        title: "Something went wrong",
+        message: error,
+      });
+    }
+  }, [error]);
+
+  async function handleSubmit(values: Values) {
+    setIsLoading(true);
+    try {
+      const data = await retrieveBrand(bfApiKey, values.textfield);
+
+      setResp(data);
+    } catch (err: unknown) {
+      setError((err as Error).message);
+    }
+
+    setIsLoading(false);
+  }
+
+  async function saveToDownloads(imageUrl: string, extName: string) {
+    const filePath = join(DOWNLOADS_DIR, randomUUID() + "." + extName);
+    try {
+      const { data } = await axios.get(imageUrl, { responseType: "arraybuffer" });
+
+      await writeFile(filePath, data);
+      await Clipboard.copy(filePath);
+
+      showToast({
+        style: Toast.Style.Success,
+        title: `Saved To Downloads!`,
+      });
+    } catch (err: unknown) {
+      setError((err as Error).message);
+    }
+  }
+
+  return resp == null ? (
+    <Form
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm onSubmit={handleSubmit} />
+        </ActionPanel>
+      }
+      isLoading={isLoading}
+    >
+      <Form.Description text="This form accepts fields required to interact with brand fetch" />
+      <Form.TextField id="textfield" title="Brand Domain" placeholder="Enter text" defaultValue="brandfetch.com" />
```

Comment:
What else could be entered here?

Reply:
this is pretty much all the info required for this api

Reply:
So the user would just enter one domain, right? Would it make sense to land on a list instead where you can select the domain? Could you maybe record a simple screencast of the flow so I can get a better understanding 🙂

Reply:
yes just one domain.
Theoritically, a user would be able to input any domain from all over the web into this extension, so we cant have this option as a list. Does that make any sense?